odoo/addons/base/models/ir_actions.py

```diff
@@ -642,7 +762,43 @@ def _run_action_object_write(self, eval_context=None):
             for field, new_value in res.items():
                 record_cached[field] = new_value
         else:
-            self.env[self.model_id.model].browse(self._context.get('active_id')).write(res)
+            starting_record = self.env[self.model_id.model].browse(self._context.get('active_id'))
+            _, _, target_records = self._traverse_path(record=starting_record)
+            target_records.write(res)
+
+    def _run_action_webhook(self, eval_context=None):
+        """Send a post request with a read of the selected field on active_id."""
+        record = self.env[self.model_id.model].browse(self._context.get('active_id'))
+        url = self.webhook_url
+        if not url:
+            raise UserError(_("I'll be happy to send a webhook for you, but you really need to give me a URL to reach out to..."))
+        vals = {
+            '_model': self.model_id.model,
+            '_id': record.id,
+            '_action': '%s(#%s)' % (self.name, self.id),
+        }
+        if self.webhook_field_ids:
+            # you might think we could use the default json serializer of the requests library
+            # but it will fail on many fields, e.g. datetime, date or binary
+            # so we use the json.dumps serializer instead with the str() function as default
+            vals.update(record.read(self.webhook_field_ids.mapped('name'), load=None)[0])
+        json_values = json.dumps(vals, sort_keys=True, default=str)
+        _logger.info("Webhook call to %s", url)
+        _logger.debug("Webhook call to %s: POST JSON data %s", url, json_values)
+        try:
+            # 'send and forget' strategy, and avoid locking the user if the webhook
+            # is slow or non-functional (we still allow for a 1s timeout so that
+            # if we get a proper error response code like 400, 404 or 500 we can log)
+            response = requests.post(url, data=json_values, headers={'Content-Type': 'application/json'}, timeout=1)
+            response.raise_for_status()
+        except requests.exceptions.ReadTimeout:
+            _logger.warning("Webhook call timed out after 1s - it may or may not have failed. "
+                            "If this happens often, it may be a sign that the system you're "
+                            "trying to reach is slow or non-functional.")
+        except requests.exceptions.RequestException as e:
+            _logger.warning("Webhook call failed: %s", e)
```

Comment:
I'm frankly still undecided, in some cases it might be really, really annoying that notifications to an external system could rollback the transaction (e.g. send a webhook on SO confirmation, the target system is down, and suddenly your ecommerce is broken).
Perhaps it should be an option, or we should store logs in a place where users could find them...